pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java

```diff
@@ -417,20 +426,55 @@ public PulsarClient getReplicationClient(String cluster) {
         });
     }
 
+    /**
+     * It creates a topic async and returns CompletableFuture. It also throttles down configured max-concurrent topic
+     * loading and puts them into queue once in-process topics are created.
+     * 
+     * @param topic persistent-topic name
+     * @return CompletableFuture<Topic>
+     * @throws RuntimeException
+     */
     private CompletableFuture<Topic> createPersistentTopic(final String topic) throws RuntimeException {
         checkTopicNsOwnership(topic);
 
+        final CompletableFuture<Topic> topicFuture = new CompletableFuture<>();
+
+        final Semaphore topicLoadSemaphore = topicLoadRequestSemaphore.get();
+
+        if (topicLoadSemaphore.tryAcquire()) {
+            CompletableFuture<Topic> future = new CompletableFuture<>();
```

Comment:
I think this could be done with just 1 future, just the same way you add the handle method to attach behavior after completing